tests/framework/results/test_context.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from vivarium.framework.results.context import ResultsContext
+from vivarium.framework.results.stratification import Stratification
+
+from .mocks import (
+    CATEGORIES,
+    NAME,
+    SOURCES,
+    sorting_hat_bad_mapping,
+    sorting_hat_serial,
+    sorting_hat_vector,
+)
+
+
+@pytest.mark.parametrize(
+    "name, sources, categories, mapper, is_vectorized",
+    [
+        (  # expected Stratification for vectorized
+            NAME,
+            SOURCES,
+            CATEGORIES,
+            sorting_hat_vector,
+            True,
+        ),
+        (  # expected Stratification for non-vectorized
+            NAME,
+            SOURCES,
+            CATEGORIES,
+            sorting_hat_serial,
+            False,
+        ),
+    ],
+)
+def test_add_stratification(name, sources, categories, mapper, is_vectorized):
+    ctx = ResultsContext()
+    ctx.add_stratification(name, sources, categories, mapper, is_vectorized)
+    expected_object = Stratification(name, sources, categories, mapper, is_vectorized)
+    assert expected_object in ctx._stratifications
+
+
+@pytest.mark.parametrize(
+    "name, sources, categories, mapper, is_vectorized, expected_exception",
+    [
+        (  # map to a category that isn't in CATEGORIES
+            NAME,
+            SOURCES,
+            CATEGORIES,
+            sorting_hat_bad_mapping,
+            False,
```

Comment:
I'm very surprised this throws an exception. How do we know when adding a stratification that its mapper function can return something not in categories? I'm pleasantly surprised if this is actually what happens.

Reply:
Yeah, they were nonsense. I don't know why they passed though. Nevertheless, they've been removed and replaced.